analysis/freq.go

```diff
@@ -55,28 +55,29 @@ func (tfs TokenFrequencies) MergeAll(remoteField string, other TokenFrequencies)
 func TokenFrequency(tokens TokenStream, arrayPositions []uint64) TokenFrequencies {
 	rv := make(map[string]*TokenFreq, len(tokens))
 
+	tls := make([]TokenLocation, len(tokens))
+	tlNext := 0
+
 	for _, token := range tokens {
-		curr, ok := rv[string(token.Term)]
+		tls[tlNext] = TokenLocation{
+			ArrayPositions: arrayPositions,
+			Start:          token.Start,
+			End:            token.End,
+			Position:       token.Position,
+		}
+
+		term := string(token.Term)
+		curr, ok := rv[term]
```

Comment:
Thanks @nuss-justin -- learning things about go from your note.  Will reset this back to the old code and push up another pull-request.